addons/account/models/account_move.py

```diff
@@ -1412,31 +1412,58 @@ def _compute_partner_credit_warning(self):
-            if the partner exceeds a credit limit (set on the company or the partner itself).
-            :param record:                  The record where the warning will appear (Invoice, Sales Order...).
-            :param updated_credit (float):  The partner's updated credit limit including the current record.
-            :param include (bool):          Whether the current record's amount is included in the warning message.
-            :return (str):                  The warning message to be showed.
+    def _build_credit_warning_message(self, record, credit_invoiced=0.0, credit_to_invoice=0.0, current_amount=0.0):
+        """Build the warning message that will be displayed in a yellow banner on top of the current record if the
+        partner exceeds a credit limit (set on the company or the partner itself).
+
+        :param record:                    The record where the warning will appear (Invoice, Sales Order...)
+        :param credit_invoiced (float):   The partner's outstanding credit amount from invoices
+        :param credit_to_invoice (float): The partner's outstanding credit amount from sales orders to invoice
+        :param current_amount (float):    The partner's outstanding credit amount from the current document
+        :return (str):                    The warning message to be showed
         """
         partner_id = record.partner_id.commercial_partner_id
-        if not partner_id.credit_limit or updated_credit <= partner_id.credit_limit:
+        total_credit = credit_invoiced + credit_to_invoice + current_amount
+        if not partner_id.credit_limit or total_credit <= partner_id.credit_limit:
             return ''
-        msg = _('%s has reached its Credit Limit of : %s\nTotal amount due ',
-                partner_id.name,
-                formatLang(self.env, partner_id.credit_limit, currency_obj=record.company_id.currency_id))
-        if include:
-            msg += _('(including this document) ')
-        msg += ': %s' % formatLang(self.env, updated_credit, currency_obj=record.company_id.currency_id)
-        return msg
+        msg = _(
+            '%(partner_name)s has reached its credit limit of: %(credit_limit)s',
+            partner_name=partner_id.name,
+            credit_limit=formatLang(self.env, partner_id.credit_limit, currency_obj=record.company_id.currency_id)
+        )
+        if credit_to_invoice > 0 and current_amount > 0:
+            return msg + '\n' + _(
+                'Total amount due (including sales orders and this document): %(total_credit)s',
+                total_credit=formatLang(self.env, total_credit, currency_obj=record.company_id.currency_id)
+            )
+        elif credit_to_invoice > 0:
+            return msg + '\n' + _(
+                'Total amount due (including sales orders): %(total_credit)s',
+                total_credit=formatLang(self.env, total_credit, currency_obj=record.company_id.currency_id)
+            )
```

Comment:
Don't forget to handle the translations :wink:

Reply:
Do I just adapt the `.pot` file? That's what I added now.

Reply:
Yep, that's fine :+1: